### PR TITLE
SONAR-10346 fix DB migration

### PR DESCRIPTION
**Pull request by sns-seb**

### CODE REVIEW

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java [hunk 1]
```diff
@@ -19,27 +19,30 @@
...
 import org.sonar.server.platform.db.migration.step.DataChange;
 import org.sonar.server.platform.db.migration.version.v63.DefaultOrganizationUuidProvider;
 
-import static com.google.common.base.Preconditions.checkNotNull;
 import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toMap;
 
 public class MigrateWebhooksToWebhooksTable extends DataChange {
 
+  private static final long NO_RESOURCE_ID = -8435121;
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java]: Code Smell Code Smell: Add underscores to this numeric value for readability ([squid:S2148](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2148))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&pullRequest=3148&issues=AWIfRM18oomSmF0fSAj0&open=AWIfRM18oomSmF0fSAj0)

[ISSUE_KEY]: AWIfRM18oomSmF0fSAj0

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java [hunk 2]
```diff
@@ -53,47 +56,112 @@ public MigrateWebhooksToWebhooksTable(Database db, DefaultOrganizationUuidProvid
...
-    Map<String, PropertyRow> rows = context
-      .prepareSelect("select id, prop_key, resource_id, text_value, created_at from properties where prop_key like 'sonar.webhooks%'")
+    Multimap<Long, PropertyRow> rows = context
+      .prepareSelect("select" +
+        " props.id, props.prop_key, props.resource_id, prj.uuid, props.text_value, props.created_at" +
+        " from properties props" +
+        " left join projects prj on prj.id = props.resource_id and prj.scope = ? and prj.qualifier = ?" +
+        " where" +
+        " props.prop_key like 'sonar.webhooks%'" +
+        " and props.text_value is not null")
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java]: you could even add `and prj.enabled=?` with value `true`.

File: server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTableTest.java [hunk 3]
```diff
@@ -202,4 +265,179 @@ private void insertProject(String organizationUuid, String projectId, String pro
...
+    private final String name;
+    private final String url;
+    @Nullable
+    private final String organizationUuid;
+    @Nullable
+    private final String projectUuid;
+    private final long createdAt;
+    private final long updatedAt;
+
+    private Row(String uuid, String name, String url, @Nullable String organizationUuid, @Nullable String projectUuid, long createdAt, long updatedAt) {
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTableTest.java]: Constructor can be dropped.

File: server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTableTest.java [hunk 4]
```diff
@@ -202,4 +265,179 @@ private void insertProject(String organizationUuid, String projectId, String pro
...
+  private static final class Webhook {
+    private final String name;
+    private final String url;
+    @Nullable
+    private final String organizationUuid;
+    @Nullable
+    private final String projectUuid;
+    private final long createdAt;
+
+    private Webhook(String name, String url, @Nullable String organizationUuid, @Nullable String projectUuid, long createdAt) {
```
Comment (simonbrandhof) [inline server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTableTest.java]: According to callers the parameter "name" can be @Nullable.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java (hunk 1 above)
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v71/MigrateWebhooksToWebhooksTable.java]: Code Smell Code Smell: Add underscores to this numeric value for readability ([squid:S2148](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2148))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&pullRequest=3148&issues=AWIfRM18oomSmF0fSAj0&open=AWIfRM18oomSmF0fSAj0)

[ISSUE_KEY]: AWIfRM18oomSmF0fSAj0

**Comment by sns-seb** [comment]
@simonbrandhof feedback applied, please review again

**Pull request merged**